### DISCUSSION TITLE
Create PDF that's editable in Canva

### DISCUSSION THREAD

**Discussion by foranuj** [pymupdf/PyMuPDF#3850] [Looking for help]
I want to create PDF that contains a series of images, a collage so to speak. However, I'd like this PDF then to be exported to Canva and opened there for further editing, like adding borders etc. Each image inside the original PDF also needs to be editable by either resizing or whatever functionality Canva provides.

I am wondering how can I achieve this, currently I'm creating my collage and then inserting the images in the corresponding locations by using something like this

`# Convert the resized PIL image to bytes for PyMuPDF
                    img_byte_arr = BytesIO()
                    img.save(img_byte_arr, format='PNG')  # Save as PNG format
                    img_byte_arr = img_byte_arr.getvalue()  # Get byte data

                    # Define the position on the PDF where the image should be placed
                    pos_x, pos_y = int(cell.x), int(cell.y)
                    img_width, img_height = img.size

                    # Insert the image into the PDF page at the correct position
                    page.insert_image(
                        fitz.Rect(pos_x, pos_y, pos_x + img_width, pos_y + img_height),
                        stream=img_byte_arr,
                        width=img_width,
                        height=img_height
                    )
`

And then I save the final PDF 

`            if self.output_file:
                pdf.save(self.output_file, deflate=False)
                pdf.close()
`

But when I open this PDF file in Canva, I can't manipulate any of the existing individual images. Should I be using embed instead?

Thanks in advance

**Comment by JorjMcKie** [comment]
I don't know how Canva works. What do you mean by "embed"? There is no such function in PyMuPDF.

**Comment by foranuj** [comment]
I was thinking this function embfile_add, and Canva is just an example, the idea is that I'd like to manipulate the PDF that contains a bunch of images on a single page in an external editor. And the external editor should be able to recognize that there's more than 1 image on that page.

**Reply by JorjMcKie** [reply]
Well you can of course add image files via embfile_add just like anything else. But this won't be displayed on any page.
`EmbeddedFiles` are stored on the PDF **_file_** level. So this won't help you here.
I have written a basic [image editor](https://github.com/pymupdf/PyMuPDF-Utilities/tree/master/examples/edit-images) some time ago based on wxPython.

**Reply by foranuj** [reply]
I see. I might have to work with a custom image editor then... Thanks.
